pkg/util/trivy/trivy.go

```diff
@@ -310,8 +310,10 @@ func (c *Collector) ScanFilesystem(ctx context.Context, path string, scanOptions
 }
 
 func (c *Collector) scan(ctx context.Context, artifact artifact.Artifact, applier applier.Applier, imgMeta *workloadmeta.ContainerImageMetadata) (*types.Report, error) {
-	if imgMeta != nil {
-		artifactReference, err := artifact.Inspect(ctx) // called by the scanner as well
+	if imgMeta != nil && c.cacheCleaner != nil {
```

Comment:
could you add a comment that explains why the cacheCleaner can be nil?

Reply:
Done